UPGRADE.rst

```diff
@@ -52,53 +52,165 @@ returned by the Client-Server API:
+
+If ``email.trust_identity_server_for_password_resets`` is ``false`` or absent
+and a threepid type in ``account_threepid_delegates`` is not set,
+then Synapse will attempt to send password reset and registration messages for
+that type.
+
+Custom templates
+---------------
 
-If ``email.trust_identity_server_for_password_resets`` is ``false`` or absent and a threepid
-type in ``account_threepid_delegates`` is not set to a domain, then Synapse will attempt to
-send password reset and registration messages for that type.
+If you have configured a custom template directory with the
+``email.template_dir`` option, be aware that there are new templates regarding
+registration and threepid management.
 
-Email templates
----------------
+``registration.html`` and ``registration.txt`` contain the content of the email
+that is sent to a user when registering with an email address.
+
+``registration_success.html`` and ``registration_failure.html`` are templates
+that will be shown to the user when they click the link in that confirmation
+email, either showing them a success page (in the case that no onward redirect
+was specified by the client), or a failure page. is not configured).
 
-If you have configured a custom template directory with the ``email.template_dir`` option, be
-aware that there are new templates regarding registration. ``registration.html`` and
-``registration.txt`` have been added and contain the content that is sent to a client upon
-registering via an email address.
+``add_threepid.html`` and  ``add_threepid.txt`` contain the content of the
+email that is sent to a user when they make a request to add an email address
+to their account.
 
-``registration_success.html`` and ``registration_failure.html`` are also new HTML templates
-that will be shown to the user when they click the link in their registration emai , either
-showing them a success or failure page (assuming a redirect URL is not configured).
+``add_threepid_failure.html`` and ``add_threepid_success.html`` are templates
+that will be shown to the user when they click the link in that confirmation
+email, either showing them a success page (in the case that no onward redirect
+was specified by the client), or a failure page.
 
-Synapse will expect these files to exist inside the configured template directory. To view the
-default templates, see `synapse/res/templates
+Synapse will expect these files to exist inside the configured template
+directory. To view the default templates, see `synapse/res/templates
 <https://github.com/matrix-org/synapse/tree/master/synapse/res/templates>`_.
 
+
+SMS
+---
+
+Following upgrade, the only way to maintain the ability to send SMS (and
+therefore register via a phone number) will be to continue to delegate SMS
+delivery via the matrix.org and vector.im identity servers (or another identity
+server that supports SMS sending).
+
+
+The ``account_threepid_delegates`` dictionary defines whether the homeserver
```

Comment:
see comments above, though I'd love it if we could avoid repeating this

Reply:
I'm just not sure I see a neat way not to repeat, since it is convenient to have the context close the description of what changes the admin must make. My sense is that since this is not a doc that needs maintenance the repetition is permissible, if you can see a better way to do it, just edit direct.